internal/pkg/cli/svc_deploy.go

```diff
@@ -424,14 +431,12 @@ func (o *deploySvcOpts) stackConfiguration(addonsURL string) (cloudformation.Sta
 	var conf cloudformation.StackConfiguration
 	switch t := mft.(type) {
 	case *manifest.LoadBalancedWebService:
-		if err := o.validateAppVersion(t); err != nil {
-			log.Errorf(`Cannot deploy service %s because the application version is incompatible.
-To upgrade the application, please run %s first (see https://aws.github.io/copilot-cli/docs/credentials/#application-credentials).
-`, aws.StringValue(t.Name),
-				color.HighlightCode("copilot app upgrade"))
-			return nil, err
-		}
 		if o.targetApp.RequiresDNSDelegation() {
+			if err := validateAlias(t, o.targetApp, o.envName, o.appVersionGetter); err != nil {
+				log.Errorf(fmtErrAliasAppVersionIncompatible, aws.StringValue(t.Name),
```

Comment:
it's not necessarily because the app version is incompatible right? it can be because of an unsupported pattern.

What do you think of the following:
```go
var errBadAliasPattern = errors.New("alias is not supported in hosted zones not managed by Copilot")

func (o *deploySvcOpts) stackConfiguration(addonsURL string) (cloudformation.StackConfiguration, error) {
  ...
  if err := validateAlias(..); err != nil {
       msg := log.Errorf(fmtErrAliasAppVersionIncompatible, ...)
       if errors.Is(err, errBadAliasPattern) {
             msg = fmt.Sprintf(`%s must match one of the following patterns:
- <name>.%s.%s.%s`,
- <name>.%s.%s,
- <name>.%s,
color.HighlightCode("http.alias"), ...)
       }
       log.Error(msg)
       return nil, err
  }
```